code/game/objects/items/weapons/grenades/custom_grenades.dm

```diff
@@ -303,3 +303,83 @@
+		to_chat(user, "<span class = 'warning'>The strike craft isn't authorized to fire near Central Command!</span>")
+		return
+
+	if((!ismindshielded(user)) || ((user.mind.special_role != SPECIAL_ROLE_ERT) && (user.mind.special_role != SPECIAL_ROLE_DEATHSQUAD) && (!allowed(user)))) //if is not mindshielded, or is not an ERT/DS AND does not have high level CC access. naval officers have access_cent_commander, but not special role so we don't want to lock them out.
+		to_chat(user, "<span class = 'warning'>Unauthorized user detected.</span>")
+	else
+		for(var/obj/item/grenade/bsa/B in world) //check if anyone else is calling a strike currently, we don't want to spam the announcements
+			if(B.active && B != src)
+				to_chat(user, "<span class = 'warning'>Another strike is already in progress!</span>")
+				return
+		active = TRUE
+
+		var/area/A = get_area(bombturf)
+		to_chat(user, "<span class = 'warning'>Calibrating GPS, remain still to continue with the strike.</span>")
+		
+		//tell admins and ghosts we are initiating a strike
+		message_admins("[key_name_admin(user)] is initiating a Bluespace Artillery Strike at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>") 
+		notify_ghosts("[user] is initiating a Bluespace Artillery Strike in [get_area(src)]!", source = src)
+		log_game("[key_name_admin(user)] is initiating a Bluespace Artillery Strike at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>")
+		
+		playsound(src, 'sound/effects/bsainit.ogg', 50, 0)
+		
+		//make sure they stay still, if they move it's cancelled
+		if(do_after(user, arm_time, target = user))
+			playsound(src, 'sound/effects/bsainbound.ogg', 70, 0, 6, 3)
+			light_power = 8
+			light_color = "red"
+			light_range = 10
+			update_light()
+			priority_announcement.Announce("This is NDV Brutus, we've received your strike signal loud and clear, [user]! Bluespace Artillery Strike incoming near [A.name]!", "Bluespace Artillery Strike", 'sound/effects/engine_alert2.ogg')
+			to_chat(user, "<span class = 'warning'>Strike sequence initiated! Deploy beacon to target! [strike_init_time] seconds!</span>")
+			caller = user
+			if(iscarbon(user))
+				var/mob/living/carbon/C = user
+				C.throw_mode_on()
+			addtimer(CALLBACK(src, .proc/prime), strike_init_time * 10)
+		else
+			active = FALSE
+			to_chat(user, "<span class = 'warning'>Calibration cancelled. Strike sequence aborted.</span>")
+			message_admins("[key_name_admin(user)] has cancelled a Bluespace Artillery Strike at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>")
+		
+		
+/obj/item/grenade/bsa/prime()
+	update_mob()
+	var/ex_power = 3
+	var/turf/bombturf = get_turf(src)
+	var/area/A = get_area(bombturf)
+	explosion(get_turf(src), ex_power * 2, ex_power * 5, ex_power * 6)
```

Comment:
Instead of defining ex_power in prime(), it should be defined at the object level, so it can be varedited.
Or, better yet, scrap the idea of using a single ex_power var, and have devastation/high/light/flame/etc power as independent vars defined at the object level, perhaps as a var/list/bsa_explosive_effects(...). Put the announcement text in a var, too.
And have the actual detonation handled by a proc/call_in_bsa defined at the object level, which can be overriden.

That would allow several things:
- easy varediting of the bsa grenade to alter its yield
- easier creation of variable power subtypes, such as a low power variant for red team commanders, or a high power variant for specops/DS
- creation of special effect subtypes, such as a type that functions as a huge-radius EMP nade ("Orbital Ion Cannon shot"), purely fire nade ("thermal beam strike"), or even other effects ("purifier ray" v cult/vampires/etc). Heck, even a syndicate version would be possible with this system (probably reserved for nukeops).

Reply:
Did some, but most of it can be saved for another PR so that it's not a massive PR with the feature chill